x/icqoracle/keeper/genesis.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	"time"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v24/x/icqoracle/types"
+)
+
+// Loads module state from genesis
+func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
+	k.SetParams(ctx, genState.Params)
+	for _, tokenPrice := range genState.TokenPrices {
+		tokenPrice.SpotPrice = math.LegacyZeroDec()
+		tokenPrice.UpdatedAt = time.Time{}
+		tokenPrice.QueryInProgress = false
```

Comment:
Sam:

i don't think you want to override these

The way I think about this function is, if we (hypothetically) ever had to fork the chain, we would want to export the full state and then initialize the new chain with that state. So in this world, I don't think we would want to mess with the fields.

Wdyt?

Reply:
Assaf:

This function is only called on genesis, where we don't even have IBC set up. In my mind these values are definitely stale at this point.

I guess there's no real reason to reset SpotPrice & UpdatedAt because there's also PriceExpirationTimeoutSec to make sure they are not used when stale. As for QueryInProgress, there's definitely no IBC at this point so I feel confident we can reset it and prevent potential issues down the road.

Reply:
> In my mind these values are definitely stale at this point.

Agreed with that

---

However, this function is also relevant when we do state exported testnets when we start up a forked chain with the current state. In this case, we would notice there are some differences (which we could visually verify are fine, but still kind of annoying to deal with).

I think it makes more sense to just keep values as what they are. The genesis comes from the JSON which is hard coded.In practice it will never be used, but in theory, if we had specific values set on the JSON, that probably means we want to use them (otherwise, we would just scrape them ourselves)

Reply:
okay, and what about `QueryInProgress`?

Reply:
same deal, my personal opinion is that if we had it set to true for some reason in genesis.json, we should keep it like that (otherwise we would have manually set it to false)